### PR TITLE
PRODENG-2550 Cleanup of the terraform module

### PR DESCRIPTION
**Pull request by dimitarbytes**
- Removed all default passwords from variables
- Now we are creating the worker and windows worker modules conditionally 

https://mirantis.jira.com/browse/PRODENG-2550

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -121,9 +123,9 @@ locals {
       role             = host.tags["Role"]
       privateInterface = "ens5"
     }
-  ]
-  windows_workers = [
-    for host in module.windows_workers.machines : {
+  ] : []
+  windows_workers = var.windows_worker_count > 0 ? [
```
Comment (james-nesbitt) [inline main.tf:127]: is this detection logic necessary?  won't the module.windows_workers list just be an empty list?

Comment (dimitarbytes) [reply main.tf:127]: all the workers/managers/msr need to be initialized to something because we are using them later to concat all the nodes in:
`      hosts = concat(local.managers, local.workers, local.msr, local.windows_workers)
`

Comment (dimitarbytes) [reply main.tf:127]: I just ran it as suggested I get the following error
`╷
│ Error: Invalid index
│ 
│   on main.tf line 128, in locals:
│  128:     for host in module.windows_workers[0].machines : {
│     ├────────────────
│     │ module.windows_workers is empty tuple
│ 
`
There is nothing to stop TF from trying to access the module at index [0]

File: variables.tf [hunk 2]
```diff
@@ -84,18 +83,18 @@ variable "msr_volume_size" {
 }
 
 variable "windows_administrator_password" {
-  default     = "w!ndozePassw0rd"
+  default     = ""
```
Comment (pgedara) [inline variables.tf:86]: @cranzy We may need to test if an empty password works, if not, we should enforce the user to provide a valid password by defining a validation for the password.

Comment (dimitarbytes) [reply variables.tf:86]: ya it does complete fine. It's just that the password field will be missing
`  "hosts":
  - "role": "manager"
    "ssh":
      "address": "3.141.35.201"
      "keyPath": "./ssh_keys/ddd-mke.pem"
      "user": "ubuntu"
  - "role": "worker"
    "winRM":
      "address": "18.225.55.111"
      "insecure": true
      "password": null
      "useHTTPS": true
      "user": "Administrator"
  - "role": "worker"
    "winRM":
      "address": "18.191.172.120"
      "insecure": true
      "password": null
      "useHTTPS": true
      "user": "Administrator"`
      
   But that's on the user to be honest.

Comment (pgedara) [reply variables.tf:86]: Resolving and we agreed to have a separate ticket to enforce windows password when windows nodes are created.

**Review by pgedara — APPROVED** [review]
